gateway/mw_auth_key.go

```diff
@@ -95,7 +97,7 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 	}
 
 	// skip auth key check if the request is looped.
-	if ses := ctxGetSession(r); ses != nil && !ctxCheckLimits(r) {
```

Comment:
I think revert this line, ctxCheckLimits remains in scope? the boolean operator change is a code smell